### PR TITLE
Add example on writing an output file of hdf5 format

### PR DESCRIPTION
**Pull request by dineshadepu**
Added an example on output of hdf5 file format. Addresses on of the checklist on #605

### CODE REVIEW

**Comment by streeve** [comment]
Also note this will need to be formatted with `clang-format`

**Comment by dineshadepu** [comment]
I have updated the file with the changes.

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+      t += dt;
+    }
+}
+
+
+int main( int argc, char *argv[] )
+{
+
+  MPI_Init( &argc, &argv );
+  Kokkos::ScopeGuard scope_guard( argc, argv );
```
[the remark this answers is not in the document]
Comment (dineshadepu) [reply example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp]: I have tried the following. However, it resulted in a bug. 

```cpp
int main( int argc, char* argv[] )
{


    Kokkos::initialize( argc, argv );
    Kokkos::ScopeGuard scope_guard( argc, argv );

    hdf5_output();

    Kokkos::finalize();

    return 0;
}
```

The bug report is as follows

```
Kokkos Error: Creating a ScopeGuard while Kokkos is initialized is illegal.
Do instead:
  std::unique_ptr<Kokkos::ScopeGuard> guard =
    !Kokkos::is_initialized() && !Kokkos::is_finalized()?
    new ScopeGuard(argc,argv) : nullptr;

Backtrace:
                  [0x55d979484859]
                  [0x55d9794783ce]
                  [0x55d97947840f]
                  [0x55d979451664]
                  [0x55d9794488b7]
                  [0x7fc1af6aad90]
__libc_start_main [0x7fc1af6aae40]
                  [0x55d9794490e5]
[CLWS00228:279839] *** Process received signal ***
[CLWS00228:279839] Signal: Aborted (6)
[CLWS00228:279839] Signal code:  (-6)
[CLWS00228:279839] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x7fc1af6c3520]
[CLWS00228:279839] [ 1] /lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c)[0x7fc1af717a7c]
[CLWS00228:279839] [ 2] /lib/x86_64-linux-gnu/libc.so.6(raise+0x16)[0x7fc1af6c3476]
[CLWS00228:279839] [ 3] /lib/x86_64-linux-gnu/libc.so.6(abort+0xd3)[0x7fc1af6a97f3]
[CLWS00228:279839] [ 4] ./example/core_tutorial/13_hdf5_output/HDF5Output(+0x42414)[0x55d979478414]
[CLWS00228:279839] [ 5] ./example/core_tutorial/13_hdf5_output/HDF5Output(+0x1b664)[0x55d979451664]
[CLWS00228:279839] [ 6] ./example/core_tutorial/13_hdf5_output/HDF5Output(+0x128b7)[0x55d9794488b7]
[CLWS00228:279839] [ 7] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90)[0x7fc1af6aad90]
[CLWS00228:279839] [ 8] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80)[0x7fc1af6aae40]
[CLWS00228:279839] [ 9] ./example/core_tutorial/13_hdf5_output/HDF5Output(+0x130e5)[0x55d9794490e5]
[CLWS00228:279839] *** End of error message ***
Aborted (core dumped)
```

So I have retained the original code. Can you please tell me how to use kokkos::initialize?

[the remark this answers is not in the document]
Comment (streeve) [reply example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp]: Right, if you use init/finalize you need to remove the `ScopeGuard`

**Comment by dineshadepu** [comment]
I have squashed the commits and made into a single commit. Updated the files as requested.

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+            /*
+              Inorder to write the particle properties to an hdf5 files, we use
+
+              Cabana::Experimental::HDF5ParticleOutput::writeTimeStep function.
+
+              First seven arguments of `writeTimeStep` are mandotory arguments
+              and rest are optional fields. The seven arguments are as follows
+
+
+              1. config file
```
Comment (brtnfld) [inline example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp:134]: Not a file, so maybe something like:

1. struct used for tuning HDF5

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp [hunk 3]
```diff
@@ -0,0 +1,180 @@
...
+              2. name of the file
```
Comment (brtnfld) [inline example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp:135]: This is the filename's prefix.

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp [hunk 4]
```diff
@@ -0,0 +1,180 @@
...
+              4. time step value (int)
+              5. time of the simulation (double)
+              6. number of particles (int)
+              7. positions (double[3])
+              8. Optional fields
+              9. Optional fields
+              10. ...
+
+              Please look at the source for more details
+              https://github.com/ECP-copa/Cabana/blob/master/core/src/Cabana_HDF5ParticleOutput.hpp#L496
```
Comment (brtnfld) [inline example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp:146]: Maybe refer to the documentation instead:

https://ecp-copa.github.io/Cabana/doxygen/Cabana__HDF5ParticleOutput_8hpp.html

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp (hunk 2 above)
Comment (streeve) [reply example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp]: Updated with both suggestions

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp (hunk 3 above)
Comment (streeve) [reply example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp]: Fixed

File: example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp (hunk 4 above)
Comment (streeve) [reply example/core_tutorial/13_hdf5_output/hdf5_output_example.cpp]: I removed for now, but I am going to look for a way to automatically link each example to the doxygen

**Comment by streeve** [comment]
Look good to you @brtnfld?

**Review by brtnfld — APPROVED** [review]
LGTM

**Review by streeve — APPROVED** [review]

**Comment by streeve** [comment]
HIP timeout unrelated

**Pull request merged**